source/infrastructure/pipeline/e2e-tests/utils/ec2-test-utils.ts

```diff
@@ -0,0 +1,28 @@
+/*****************************************************************************
+ *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.   *
+ *                                                                            *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may   *
+ *  not use this file except in compliance with the License. A copy of the    *
+ *  License is located at                                                     *
+ *                                                                            *
+ *      http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                            *
+ *  or in the 'license' file accompanying this file. This file is distributed *
+ *  on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,        *
+ *  express or implied. See the License for the specific language governing   *
+ *  permissions and limitations under the License.                            *
+ *****************************************************************************/
+import * as ec2 from "@aws-sdk/client-ec2";
+
+export async function getInstanceState(client: ec2.EC2Client, instanceId: string) {
+  const result = await client.send(
+    new ec2.DescribeInstanceStatusCommand({
+      InstanceIds: [instanceId],
+      IncludeAllInstances: true
+    })
+  )
+
+  if(!result.InstanceStatuses) throw new Error(`Instance with id of ${instanceId} Not Found`)
+
+  return result.InstanceStatuses![0].InstanceState!.Name
```

Comment:
Is assert not null correct here? If there's no response, we should probably throw an error. Though I guess that will happen anyway when you try to access a nonexistent property. You also already check, so I'm not sure why the assertion is necessary.

Reply:
the bang assert is there to control the typing. For unit-test code I typically want it to be as simple and straightforward as possible, and any errors that happen can just cause the test to fail and print to console. No need to waste the cognitive complexity on extra error handling

Reply:
You are getting some kind of error handling either way. With bang you are just saying you're happy with errors about accessing properties of undefined. It feels like an abuse of the type system to me. For cases like this, you may get behavior you're happy with, but for values you pass around that have types that aren't strictly correct, you might run into strange situations later when a `string` results in a weird exception because it's not a string.

Reply:
My assumption is that I am asserting the shape of a very specific return element from a larger api that can be used in other ways. I do not expect this value to ever be undefined except when the instanceID does not match up to a existing test instance. I could certainly be wrong about the behavior of the underlying interface, but I am basically attempting to take an interface build for batching and use it as a single-item lookup.